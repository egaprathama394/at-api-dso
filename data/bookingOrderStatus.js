export default {
  orderDaimoCode: 'DFA202000000123',
  pssOrderStatus: '2',
};
