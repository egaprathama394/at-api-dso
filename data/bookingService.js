const randomstring = require('randomstring');

const fullNameMax = randomstring.generate({
  charset: 'string',
  length: '110',
});

const noPhoneMax = randomstring.generate({
  charset: 'numeric',
  length: '17',
});

const noPhoneRandom = randomstring.generate({
  charset: 'numeric',
  length: '18',
});

const fullNameRandom = randomstring.generate({
  charset: 'string',
  length: '111',
});

const randomNopol1 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol2 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol3 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol4 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

export default {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AB${randomNopol1}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght

export const maxInput = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameMax}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data booking service dengan inputan max char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneMax}`,
  policeRegNo: `AC${randomNopol2}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght + 1

export const negativeRequest = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameRandom}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service dengan inputan max + 1 char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneRandom}`,
  policeRegNo: `AD${randomNopol3}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for cancel booking service

export const inputDataForCancel = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data untuk cancel booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AA${randomNopol4}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const cancelRequest = {
  orderNo: '',
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

// data for display booking service

export const displayBookingService = {
  isWalkIn: false,
  orderBy: '',
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const invalidDisplayBookingService = {
  isWalkIn: '',
  orderBy: '',
  AccountType: '',
  Action: '',
  ApplicationName: '',
  DeviceType: '',
  OperatingSystem: '',
  UserVersion: '',
};
