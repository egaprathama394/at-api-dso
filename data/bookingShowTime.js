export default {
  branchCode: 'D004',
  scheduleDate: '2020-09-30 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const negativeRequest = {
  branchCode: 'D004',
  scheduleDate: '2018-10-26 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
