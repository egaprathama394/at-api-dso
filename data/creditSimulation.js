export default {
  P_NEW_USED: 'N',
  P_CATEGORY: 'EX2',
  P_VEHICLE_CATEGORY: '1',
  P_BRAND: '002',
  P_TYPE: '1',
  P_MODEL: '1',
  P_DP: '49125000',
  P_OTR: '196500000',
  P_TYPE_INSU: 'A',
  P_MODE_INSU: 'K',
  P_FLAG_ACP: 'Y',
  P_TENORVALUE: '12',
};