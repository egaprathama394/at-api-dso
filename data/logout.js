export default {
  applicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  deviceID: '8B55A4DA-5331-47F9-9C8D-838A110FA18C',
};


export const negativeRequest = {
  action: 'Display',
  applicationName: '9093D285-5263-4B9A-97D7-C185189B0695',
  userVersion: '1.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
