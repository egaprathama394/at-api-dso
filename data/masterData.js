
export default {
  companyCode: '0003',
  masterDataInfo: [
    {
      lastSync: '1900-01-01 00:00:00',
      objectName: 'Branch',
    },
  ],
  vehicleBrandCode: 'DAIHATSU',
  accountType: '1',
  action: '',
  deviceType: '0',
  operatingSystem: '0',
  serVersion: '1.0.0',
};

export const JobServiceCompany = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'JobService_Company',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const VehicleBrand = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'vehicleBrand',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const district = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'district',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const province = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'province',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const region = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'region',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

export const holidayCalendar = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'holidayCalendar',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',

};

// looping data for job service company

export const loopVehicleType = [
  {
    vehicleType: 'AYLA',
    title: 'test case get master data job service company where vehicle = AYLA',
  },
  {
    vehicleType: 'GRANMAX',
    title: 'test case get master data job service company where vehicle = GRANMAX',
  },
  {
    vehicleType: 'GRANMAX1.3',
    title: 'test case get master data job service company where vehicle = GRANMAX1.3',
  },
  {
    vehicleType: 'GRANMAX1.5',
    title: 'test case get master data job service company where vehicle = GRANMAX1.5',
  },
  {
    vehicleType: 'LUXIO',
    title: 'test case get master data job service company where vehicle = LUXIO',
  },
  {
    vehicleType: 'Sigra 1.0',
    title: 'test case get master data job service company where vehicle = Sigra 1.0',
  },
  {
    vehicleType: 'Sigra 1.2',
    title: 'test case get master data job service company where vehicle = Sigra 1.2',
  },
  {
    vehicleType: 'SIRION',
    title: 'test case get master data job service company where vehicle = SIRION',
  },
  {
    vehicleType: 'TERIOS',
    title: 'test case get master data job service company where vehicle = TERIOS',
  },
  {
    vehicleType: 'XENIA1.0',
    title: 'test case get master data job service company where vehicle = XENIA1.0',
  },
  {
    vehicleType: 'XENIA1.3',
    title: 'test case get master data job service company where vehicle = XENIA1.3',
  },
  {
    vehicleType: 'OTHERS',
    title: 'test case get master data job service company where vehicle = OTHERS',
  },
];

export const VehicleTestDrive = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'VehicleTestDrive',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const parameterConfigApp = {
  vehicleBrandCode: 'DAIHATSU',
  companyCode: '0003',
  masterDataInfo: [
    {
      objectName: 'ParameterConfigApplication',
      lastSync: '1900-06-17 00:00:00',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
