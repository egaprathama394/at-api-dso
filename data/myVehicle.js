const randomstring = require('randomstring');

const randomNopol = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomVin = randomstring.generate({
  charset: 'string',
  length: '17',
});

const randomMaxVin = randomstring.generate({
  charset: 'string',
  length: '18',
});

const randomPlusVin = randomstring.generate({
  charset: 'string',
  length: '19',
});

const randomNopolPlus = randomstring.generate({
  charset: 'string',
  length: '8',
});

const randomNopolMax = randomstring.generate({
  charset: 'string',
  length: '7',
});
// Data for display all My Vehicle

export default {
  action: 'DisplayAll',
  vehicleBrand: 'DAIHATSU',
  applicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const negativeRequest = {
  action: 'DisplayAll',
  vehicleBrand: 'xxxx',
  applicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

// Data for add my vehicle

export const addMyVehicle = {
  MyVehicle: {
    id: 0,
    isConfirmed: false,
    isDefault: false,
    policeRegNo: `B${randomNopol}TEST`,
    stnkDate: '2019-10-22 11:06:40',
    vehicleBrand: 'DAIHATSU',
    vehicleName: 'LUXIO',
    vehicleType: 'LUXIO',
    year: 2014,
    VIN: `${randomVin}`,
  },
  VehicleBrand: 'Daihatsu',
  AccountType: '1',
  Action: 'Add',
  ApplicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const addMyVehicleNegative = {
  MyVehicle: {
    id: 0,
    isConfirmed: false,
    isDefault: false,
    policeRegNo: 'B000COR',
    stnkDate: '2019-10-22 11:06:40',
    vehicleBrand: 'DAIHATSU',
    vehicleName: 'LUXIO',
    vehicleType: 'LUXIO',
    year: 2014,
  },
  VehicleBrand: 'Daihatsu',
  AccountType: '1',
  Action: 'Add',
  ApplicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const addMyVehicleMaxInput = {
  MyVehicle: {
    id: 0,
    isConfirmed: false,
    isDefault: false,
    policeRegNo: `B0000${randomNopolMax}`,
    stnkDate: '2019-10-22 11:06:40',
    vehicleBrand: 'DAIHATSU',
    vehicleName: 'AYLA',
    vehicleType: 'AYLA',
    year: 2014,
    VIN: `${randomMaxVin}`,
  },
  VehicleBrand: 'Daihatsu',
  AccountType: '1',
  Action: 'Add',
  ApplicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// Data for add my Vehicle using max lenght + 1 char at nopol and VIN
export const addMyVehiclePlus = {
  MyVehicle: {
    id: 0,
    isConfirmed: false,
    isDefault: false,
    policeRegNo: `B0000${randomNopolPlus}`,
    stnkDate: '2019-10-22 11:06:40',
    vehicleBrand: 'DAIHATSU',
    vehicleName: 'AYLA',
    vehicleType: 'AYLA',
    year: 2014,
    VIN: `${randomPlusVin}`,
  },
  VehicleBrand: 'Daihatsu',
  AccountType: '1',
  Action: 'Add',
  ApplicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// Data fo delete my vehicle

export const deleteMyVehicle = {
  action: 'Remove',
  applicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  userVersion: '1.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  myVehicle: {
    id: '',
  },
};

export const deletenegativeRequest = {
  action: 'Remove',
  applicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  userVersion: '1.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  myVehicle: {
    id: '7777',
  },
};

export const deleteNotExistVehicle = {
  action: 'Remove',
  applicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  userVersion: '1.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  myVehicle: {
    id: '0001',
  },
};

// Data for edit my vehicle

export const editMyVehicle = {
  MyVehicle: '',
  VehicleBrand: 'Daihatsu',
  AccountType: '1',
  Action: 'Edit',
  ApplicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// add my vehicle for delete test

export const addForDeleteMyVehicle = {
  MyVehicle: {
    id: 0,
    isConfirmed: false,
    isDefault: false,
    policeRegNo: 'B5555TEST',
    stnkDate: '2019-10-22 11:06:40',
    vehicleBrand: 'DAIHATSU',
    vehicleName: 'LUXIO',
    vehicleType: 'LUXIO',
    year: 2014,
    VIN: '1234567890abcdefg',
  },
  VehicleBrand: 'Daihatsu',
  AccountType: '1',
  Action: 'Add',
  ApplicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};
