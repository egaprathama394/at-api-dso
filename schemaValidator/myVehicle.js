export default {
  type: 'object',
  required: [
    'userVehicleList',
    'acknowledge',
    'version',
    'build',
    'errorCode',
  ],
  properties: {
    userVehicleList: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'id',
          'vehicleType',
          'policeRegNo',
          'engineNo',
          'vin',
          'year',
          'stnkDate',
          'vehicleName',
          'vehicleBrand',
          'urlPhoto',
          'isDefault',
          'isConfirmed',
          'printFakturDate',
          'lastServiceDate',
        ],
        properties: {
          id: {
            type: 'integer',
          },
          vehicleType: {
            type: 'string',
          },
          policeRegNo: {
            type: 'string',
          },
          engineNo: {
            type: 'string',
          },
          vin: {
            type: 'string',
          },
          year: {
            type: 'integer',
          },
          stnkDate: {
            type: 'string',
          },
          vehicleName: {
            type: 'string',
          },
          vehicleBrand: {
            type: 'string',
          },
          urlPhoto: {
            type: 'string',
          },
          isDefault: {
            type: 'boolean',
          },
          isConfirmed: {
            type: 'boolean',
          },
          printFakturDate: {
            type: 'string',
          },
          lastServiceDate: {
            type: 'string',
          },
        },
      },
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'integer',
    },
  },
  additionalProperties: false,
};

export const negativeRequestSchema = {
  type: 'object',
  required: [
    'acknowledge',
    'message',
    'version',
    'build',
    'errorCode',
  ],
  properties: {
    acknowledge: {
      type: 'integer',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'integer',
    },
  },
};

export const AddMyVehicleSchema = {
  properties: {
    userVehicleList: {
      type: 'array',
    },
    id: {
      type: 'number',
    },
    acknowledge: {
      type: 'number',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'number',
    },
  },
  required: [
    'userVehicleList',
    'id',
    'acknowledge',
    'message',
    'version',
    'build',
    'errorCode',
  ],
  additionalProperties: false,
};

export const editMyVehicleSchema = {
  properties: {
    acknowledge: {
      type: 'number',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'number',
    },
  },
  required: [
    'acknowledge',
    'message',
    'version',
    'build',
    'errorCode',
  ],
  additionalProperties: false,
};

export const deleteMyVehicleSchema = {
  type: 'object',
  required: [
    'acknowledge',
    'message',
    'version',
    'build',
    'errorCode',
  ],
  properties: {
    acknowledge: {
      type: 'integer',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'integer',
    },
  },
};
