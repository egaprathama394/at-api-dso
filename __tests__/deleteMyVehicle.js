import getVehicle from '../endpoints/myVehicle';
import * as myVehicleSchema from '../schemaValidator/myVehicle';
import * as body from '../data/myVehicle';

const fs = require('fs');
const { matchers } = require('jest-json-schema');

expect.extend(matchers);

const ambilToken = fs.readFileSync('./token.json');
const objToken = JSON.parse(ambilToken);

const token = objToken.edit;

describe('Delete My Vehicle', () => {
  test('Can removed vehicle successfully using authorized token - positive test cases', async () => {
    const response = await getVehicle(token, body.addForDeleteMyVehicle);
    expect(response.status).toEqual(200);
    expect(response.body).toMatchSchema(myVehicleSchema.AddMyVehicleSchema);
    expect(response.body.message).toEqual(`Add vehicle VIN:${body.addForDeleteMyVehicle.MyVehicle.VIN}, PoliceRegNo:${body.addForDeleteMyVehicle.MyVehicle.policeRegNo} is success`);
    const idVehicle = await response.body.id;
    body.deleteMyVehicle.myVehicle.id = idVehicle;
    const responseDelete = await getVehicle(token, body.deleteMyVehicle);
    expect(responseDelete.status).toEqual(200);
    expect(responseDelete.body).toMatchSchema(myVehicleSchema.deleteMyVehicleSchema);
    Reflect.deleteProperty(responseDelete.body, 'build');
    expect(responseDelete.body).toMatchSnapshot();
  });
  test('can not remove vehicle that already removed using authorized token - negative test case', async () => {
    const responseDelete = await getVehicle(token, body.deletenegativeRequest);
    expect(responseDelete.status).toEqual(200);
    expect(responseDelete.body).toMatchSchema(myVehicleSchema.negativeRequestSchema);
    Reflect.deleteProperty(responseDelete.body, 'build');
    expect(responseDelete.body).toMatchSnapshot();
  });
  test('can not remove my vehicle that does not exist using authorized token - negative test cases', async () => {
    const responseDelete = await getVehicle(token, body.deleteNotExistVehicle);
    expect(responseDelete.status).toEqual(200);
    expect(responseDelete.body).toMatchSchema(myVehicleSchema.negativeRequestSchema);
    Reflect.deleteProperty(responseDelete.body, 'build');
    expect(responseDelete.body).toMatchSnapshot();
  });
});
