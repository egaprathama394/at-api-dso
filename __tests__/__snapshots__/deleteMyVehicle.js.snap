// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete My Vehicle Can removed vehicle successfully using authorized token - positive test cases 1`] = `
Object {
  "acknowledge": 1,
  "errorCode": 0,
  "message": "Remove vehicle VIN:1234567890abcdefg, PoliceRegNo:B5555TEST is success",
  "version": "1.0",
}
`;

exports[`Delete My Vehicle can not remove my vehicle that does not exist using authorized token - negative test cases 1`] = `
Object {
  "acknowledge": 0,
  "errorCode": 0,
  "message": "Vehicle is not exist",
  "version": "1.0",
}
`;

exports[`Delete My Vehicle can not remove vehicle that already removed using authorized token - negative test case 1`] = `
Object {
  "acknowledge": 0,
  "errorCode": 0,
  "message": "Vehicle is not exist",
  "version": "1.0",
}
`;
